### DISCUSSION TITLE
Onboarding not occurring

### DISCUSSION THREAD

**Discussion by ghost** [renovatebot/renovate#12491] [Request Help]
I'm not getting any on-boarding for project https://github.com/MartijnHoekstraZivver/renovate-minimize

Did I set it up incorrectly somehow? The repository started with a renovate.json file, which I removed to entice an onboarding PR.

**Comment by HonkingGoose** [comment]
Hello! 👋

Please read our installation guide [^installation], and check if you've done everything correctly.
I also recommend you read our troubleshooting guide [^troubleshooting] as well.

If after reading those pages again, you still need help, explain _how_ you're running Renovate (GitHub hosted app, self-hosted), and what you tried yourself to fix the problem.

[^installation]: https://docs.renovatebot.com/getting-started/installing-onboarding/
[^troubleshooting]: https://docs.renovatebot.com/troubleshooting/

Good luck! 😉

**Comment by rarkins** [comment]
What do the logs say? (log in at app.renovatebot.com)

**Reply by ghost** [reply]
<details>
<summary>full log</summary>

```
DEBUG: No dangling containers to remove
INFO: Repository started
{
  "renovateVersion": "28.22.3"
}
DEBUG: Using localDir: /mnt/renovate/gh/MartijnHoekstraZivver/renovate-minimize
DEBUG: Repository cache is valid
DEBUG: initRepo("MartijnHoekstraZivver/renovate-minimize")
DEBUG: Overriding default GitHub endpoint
{
  "endpoint": "https://api.github.com/"
}
DEBUG: MartijnHoekstraZivver/renovate-minimize default branch = main
DEBUG: Using app token for git init
DEBUG: resetMemCache()
DEBUG: Resetting npmrc
DEBUG: detectSemanticCommits()
DEBUG: checkOnboarding()
DEBUG: isOnboarded()
DEBUG: Checking cached config file name
DEBUG: GitHub 404
{
  "url": "repos/MartijnHoekstraZivver/renovate-minimize/contents/renovate.json"
}
DEBUG: Existing config file no longer exists
DEBUG: findFile(renovate.json)
DEBUG: Initializing git repository into /mnt/renovate/gh/MartijnHoekstraZivver/renovate-minimize
DEBUG: Performing blobless clone
DEBUG: git clone completed
{
  "durationMs": 1296
}
DEBUG: latest repository commit
{
  "latestCommit": {
    "hash": "92986184630625a0d367e1b92c7cd588c65ba695",
    "date": "2021-11-02T16:22:56+01:00",
    "message": "Delete renovate.json",
    "refs": "HEAD -> main, origin/main, origin/HEAD",
    "body": "",
    "author_name": "MartijnHoekstraZivver",
    "author_email": "90897333+MartijnHoekstraZivver@users.noreply.github.com"
  }
}
DEBUG: findFile(renovate.json5)
DEBUG: findFile(.github/renovate.json)
DEBUG: findFile(.github/renovate.json5)
DEBUG: findFile(.gitlab/renovate.json)
DEBUG: findFile(.gitlab/renovate.json5)
DEBUG: findFile(.renovaterc)
DEBUG: findFile(.renovaterc.json)
DEBUG: config file not found
DEBUG: findPr(renovate/configure, Configure Renovate, !open)
DEBUG: Retrieving PR list
DEBUG: Retrieved 0 Pull Requests
DEBUG: Found no closed onboarding PR
DEBUG: Repo is not onboarded
DEBUG: getBranchPr(renovate/configure)
DEBUG: findPr(renovate/configure, undefined, open)
DEBUG: findPr(renovate/configure, undefined, closed)
DEBUG: Onboarding PR does not exist
DEBUG: Checking if this org/owner has a default Renovate preset which can be used.
DEBUG: GitHub 404
{
  "url": "https://api.github.com/repos/MartijnHoekstraZivver/renovate-config/contents/default.json"
}
DEBUG: Failed to retrieve default.json from repo
{
  "statusCode": 404,
  "url": "https://api.github.com/repos/MartijnHoekstraZivver/renovate-config/contents/default.json"
}
INFO: Fallback to renovate.json file as a preset is deprecated, please use a default.json file instead.
DEBUG: GitHub 404
{
  "url": "https://api.github.com/repos/MartijnHoekstraZivver/renovate-config/contents/renovate.json"
}
DEBUG: Failed to retrieve renovate.json from repo
{
  "statusCode": 404,
  "url": "https://api.github.com/repos/MartijnHoekstraZivver/renovate-config/contents/renovate.json"
}
DEBUG: GitHub 404
{
  "url": "https://api.github.com/repos/MartijnHoekstraZivver/.github/contents/renovate-config.json"
}
DEBUG: Failed to retrieve renovate-config.json from repo
{
  "statusCode": 404,
  "url": "https://api.github.com/repos/MartijnHoekstraZivver/.github/contents/renovate-config.json"
}
DEBUG: No default org/owner preset found, so the default onboarding config will be used instead. Note: do not be concerned with any 404 messages that preceded this.
DEBUG: onboarding config
{
  "config": {
    "extends": [
      "config:base"
    ]
  }
}
DEBUG: No renovate config file found
DEBUG: migrateAndValidate()
DEBUG: No config migration necessary
DEBUG: massaged config
{
  "config": {
    "extends": [
      "config:base"
    ]
  }
}
DEBUG: migrated config
{
  "config": {
    "extends": [
      "config:base"
    ]
  }
}
DEBUG: Setting hostRules from config
DEBUG: Found repo ignorePaths
{
  "ignorePaths": [
    "**/node_modules/**",
    "**/bower_components/**",
    "**/vendor/**",
    "**/examples/**",
    "**/__tests__/**",
    "**/test/**",
    "**/tests/**",
    "**/__fixtures__/**"
  ]
}
DEBUG: Using file match: (^|/)tasks/[^/]+\.ya?ml$ for manager ansible
DEBUG: Using file match: (^|/)requirements\.ya?ml$ for manager ansible-galaxy
DEBUG: Using file match: (^|/)galaxy\.ya?ml$ for manager ansible-galaxy
DEBUG: Using file match: azure.*pipelines?.*\.ya?ml$ for manager azure-pipelines
DEBUG: Using file match: (^|/)batect(-bundle)?\.yml$ for manager batect
DEBUG: Using file match: (^|/)batect$ for manager batect-wrapper
DEBUG: Using file match: (^|/)WORKSPACE(|\.bazel)$ for manager bazel
DEBUG: Using file match: \.bzl$ for manager bazel
DEBUG: Using file match: (^|/)\.?bitbucket-pipelines\.ya?ml$ for manager bitbucket-pipelines
DEBUG: Using file match: buildkite\.ya?ml for manager buildkite
DEBUG: Using file match: \.buildkite/.+\.ya?ml$ for manager buildkite
DEBUG: Using file match: (^|/)Gemfile$ for manager bundler
DEBUG: Using file match: \.cake$ for manager cake
DEBUG: Using file match: (^|/)Cargo.toml$ for manager cargo
DEBUG: Using file match: (^|/).circleci/config.yml$ for manager circleci
DEBUG: Using file match: (^|/)cloudbuild.ya?ml for manager cloudbuild
DEBUG: Using file match: (^|/)Podfile$ for manager cocoapods
DEBUG: Using file match: (^|/)([\w-]*)composer.json$ for manager composer
DEBUG: Using file match: (^|/)deps\.edn$ for manager deps-edn
DEBUG: Using file match: (^|/)docker-compose[^/]*\.ya?ml$ for manager docker-compose
DEBUG: Using file match: (^|/|\.)Dockerfile$ for manager dockerfile
DEBUG: Using file match: (^|/)Dockerfile\.[^/]*$ for manager dockerfile
DEBUG: Using file match: (^|/).drone.yml$ for manager droneci
DEBUG: Using file match: (^|/).gitmodules$ for manager git-submodules
DEBUG: Using file match: ^(workflow-templates|\.github\/workflows)\/[^/]+\.ya?ml$ for manager github-actions
DEBUG: Using file match: (^|\/)action\.ya?ml$ for manager github-actions
DEBUG: Using file match: \.gitlab-ci\.yml$ for manager gitlabci
DEBUG: Using file match: \.gitlab-ci\.yml$ for manager gitlabci-include
DEBUG: Using file match: (^|/)go.mod$ for manager gomod
DEBUG: Using file match: \.gradle(\.kts)?$ for manager gradle
DEBUG: Using file match: (^|\/)gradle\.properties$ for manager gradle
DEBUG: Using file match: (^|\/)gradle\/.+\.toml$ for manager gradle
DEBUG: Using file match: \.versions\.toml$ for manager gradle
DEBUG: Using file match: (^|/)gradle/wrapper/gradle-wrapper.properties$ for manager gradle-wrapper
DEBUG: Using file match: (^|/)requirements\.yaml$ for manager helm-requirements
DEBUG: Using file match: (^|/)values.yaml$ for manager helm-values
DEBUG: Using file match: (^|/)helmfile.yaml$ for manager helmfile
DEBUG: Using file match: (^|/)Chart.yaml$ for manager helmv3
DEBUG: Using file match: ^Formula/[^/]+[.]rb$ for manager homebrew
DEBUG: Using file match: \.html?$ for manager html
DEBUG: Using file match: (^|/)plugins\.(txt|ya?ml)$ for manager jenkins
DEBUG: Using file match: (^|/)kustomization\.yaml for manager kustomize
DEBUG: Using file match: (^|/)project\.clj$ for manager leiningen
DEBUG: Using file match: \.pom\.xml$ for manager maven
DEBUG: Using file match: (^|/)pom\.xml$ for manager maven
DEBUG: Using file match: (^|/)package.js$ for manager meteor
DEBUG: Using file match: (^|/)mix\.exs$ for manager mix
DEBUG: Using file match: (^|/).node-version$ for manager nodenv
DEBUG: Using file match: (^|/)package.json$ for manager npm
DEBUG: Using file match: \.(?:cs|fs|vb)proj$ for manager nuget
DEBUG: Using file match: \.(?:props|targets)$ for manager nuget
DEBUG: Using file match: \.config\/dotnet-tools\.json$ for manager nuget
DEBUG: Using file match: (^|//)global\.json$ for manager nuget
DEBUG: Using file match: (^|/)\.nvmrc$ for manager nvm
DEBUG: Using file match: (^|/)([\w-]*)requirements\.(txt|pip)$ for manager pip_requirements
DEBUG: Using file match: (^|/)setup.py$ for manager pip_setup
DEBUG: Using file match: (^|/)Pipfile$ for manager pipenv
DEBUG: Using file match: (^|/)pyproject\.toml$ for manager poetry
DEBUG: Using file match: (^|/)\.pre-commit-config\.yaml$ for manager pre-commit
DEBUG: Using file match: (^|/)pubspec\.ya?ml$ for manager pub
DEBUG: Using file match: (^|/).python-version$ for manager pyenv
DEBUG: Using file match: (^|/)\.ruby-version$ for manager ruby-version
DEBUG: Using file match: \.sbt$ for manager sbt
DEBUG: Using file match: project/[^/]*.scala$ for manager sbt
DEBUG: Using file match: (^|/)setup\.cfg$ for manager setup-cfg
DEBUG: Using file match: (^|/)Package\.swift for manager swift
DEBUG: Using file match: \.tf$ for manager terraform
DEBUG: Using file match: (^|/)\.terraform-version$ for manager terraform-version
DEBUG: Using file match: (^|/)terragrunt\.hcl$ for manager terragrunt
DEBUG: Using file match: (^|/)\.terragrunt-version$ for manager terragrunt-version
DEBUG: Using file match: ^.travis.yml$ for manager travis
DEBUG: Matched 2 file(s) for manager sbt: build.sbt, project/Dependencies.scala
DEBUG: Found 0 package file(s)
INFO: Repository has no package files - skipping
DEBUG: Repository timing splits (milliseconds)
{
  "splits": {},
  "total": 6924
}
DEBUG: http statistics
{
  "urls": {
    "https://api.github.com/graphql (POST)": 1,
    "https://api.github.com/repos/MartijnHoekstraZivver/renovate-minimize/pulls (GET)": 1
  },
  "hostStats": {
    "api.github.com": {
      "requestCount": 2,
      "requestAvgMs": 207,
      "queueAvgMs": 1
    }
  },
  "totalRequests": 2
}
INFO: Repository finished
{
  "durationMs": 6924
}
```

</details>

What stands out to me is 

```
DEBUG: Matched 2 file(s) for manager sbt: build.sbt, project/Dependencies.scala
DEBUG: Found 0 package file(s)
INFO: Repository has no package files - skipping
```

it does find the two files that contain the dependencies, but these are not counted as package files, which makes the run fail?

I searched the documentation for "package file" but I can't figure out what a package file is, and how I can make one.

**Reply by rarkins** [reply]
I've changed a setting for your repo so that an onboarding PR is forced, but that's still not going to change the "no package files" result.

We might have a bug or a missing feature for SBT parsing. I suggest you look at the test fixtures and can maybe guess what makes yours different: https://github.com/renovatebot/renovate/tree/main/lib/manager/sbt/__fixtures__

**Reply by ghost** [reply]
It's onboarding now, but like you said, it's not picking up any dependencies (yet).

I'll see if I can debug *that* a bit further locally.

**Reply by martijnhoekstra** [reply]
The above user account was an alt account of mine, which I've now nuked. I found a fix for one of the dependencies, by changing a Seq constructor in a List constructor. Maybe that means that renovate looks specifically for the Seq constructor? Anyway, the repo is also nuked, but that seems easy to minimize.

**Reply by rarkins** [reply]
If you have a valid pattern which you think Renovate should support out of the box, please create a simple reproduction in a public repo (which you don't plan to nuke in any hurry) and then then create a feature request issue for that. It's not our intention to "force" people to only use a certain way of configuring, unless certain ways are extremely hard to parse

**Reply by martijnhoekstra** [reply]
I gave it a try at https://github.com/renovatebot/renovate/pull/12541/files, which should at least get some of the depdencies of https://github.com/martijnhoekstra/renovate-minimize

I'm not entirely sure what I'm doing as I don't fully understand how these regexes determine which seqs are supported, and in what context they're processed.
